lib/ProductOpener/Display.pm

```diff
@@ -395,28 +395,34 @@ sub analyze_request($)
 		
 	print STDERR "analyze_request : query_string 1 : $request_ref->{query_string} \n";
 	
+	# Process API parameters: fields, formats, revision
+	
 	# API calls may request JSON, JSONP or XML by appending .json, .jsonp or .xml at the end of the query string
 	# .jqm returns results in HTML specifically formated for the OFF mobile app (which uses jquerymobile)
 	# for calls to /cgi/ actions (e.g. search.pl), the format can also be indicated with a parameter &json=1 &jsonp=1 &xml=1 &jqm=1
 	# (or ?json=1 if it's the first parameter)
 	
-	# first check for the rev parameter (revision of a product)
-	
-	foreach my $parameter ('rev', 'json', 'jsonp', 'jqm','xml') {
+	# first check parameters in the query string
+
+	foreach my $parameter ('fields', 'rev', 'json', 'jsonp', 'jqm','xml') {
 	
-		if ($request_ref->{query_string} =~ /(\&|\?)$parameter=(\d+)/) {
-			$request_ref->{query_string} =~ s/(\&|\?)$parameter=(\d+)//;
+		if ($request_ref->{query_string} =~ /(\&|\?)$parameter=([^\&]+)/) {
+			$request_ref->{query_string} =~ s/(\&|\?)$parameter=([^\&]+)//;
 			$request_ref->{$parameter} = $2;
```

Comment:
Could this cause any problem if `$request_ref->{json}` were set to a non-numeric value? The check is in https://github.com/openfoodfacts/openfoodfacts-server/blob/16d18d199414fc430ce44acc51784cc54e7cb4c1/lib/ProductOpener/Display.pm#L4563 and could have to try evaluate the string as boolean value?

Reply:
I think any non-numeric value would be treated as a true value